larray/viewer.py

```diff
@@ -1604,6 +1648,103 @@ def map_filtered_to_global(self, k):
         return tuple(dkey[axis.name] for axis in self.la_data.axes)
 
 
+class _LazyLabels(object):
+    def __init__(self, arrays):
+        self.prod = Product(arrays)
+
+    def __getitem__(self, key):
+        return ' '.join(self.prod[key])
+
+    def __len__(self):
+        return len(self.prod)
+
+
+class _LazyDimLabels(object):
+    def __init__(self, prod, i):
+        self.prod = prod
+        self.i = i
+
+    def __iter__(self):
+        return iter(self.prod[:][self.i])
+
+    def __getitem__(self, key):
+        return self.prod[key][self.i]
+
+    def __len__(self):
+        return len(self.prod)
+
+
+class _LazyRange(object):
+    def __init__(self, length, offset):
+        self.length = length
+        self.offset = offset
+
+    def __getitem__(self, key):
+        if key >= self.offset:
+            return key - self.offset
+        else:
+            return ''
+
+    def __len__(self):
+        return self.length + self.offset
+
+
+class _LazyNone(object):
+    def __init__(self, length):
+        self.length = length
+
+    def __getitem__(self, key):
+        return ' '
+
+    def __len__(self):
+        return self.length
+
+
+def ndarray_to_array_and_labels(data):
+    """Converts an  Numpy ndarray into a 2D data array and x/y labels.
+
+    Parameters
+    ----------
+    data : numpy.ndarray
+        Input array.
+
+    Returns
+    -------
+    data : 2D array
+        Content of input array is returned as 2D array.
+    xlabels : 1D array
+        Labels of rows.
+    ylabels : 2D array
+        Labels of columns (cartesian product of of all axes
+        except the last one).
+    """
+    assert isinstance(data, np.ndarray)
+
+    if data.ndim == 0:
+        data.shape = (1, 1)
+        xlabels = [["{0}"], [0]]
+        ylabels = [[]]
+    elif data.ndim == 1:
+        data = data.reshape(1, data.shape[0])
+        xlabels = [["{0}"], list(range(data.size))]
```

Comment:
isn't ndim==1 just a special case of ndim >=1 regarding xlabels & ylabels (ie can we reuse the same code?)